addon/bootstrap.js

```diff
@@ -83,13 +83,21 @@ async function startup(addonData, reason) {
   // 2. sets activeExperiments in telemetry environment.
   await studyUtils.startup({ reason });
 
-  // if you have code to handle expiration / long-timers, it could go here
-  (function fakeTrackExpiration() {
-  })();
+  // log what the study variation and other info is.
+  log.debug(`info ${JSON.stringify(studyUtils.info())}`);
 
   // initiate the chrome-privileged part of the study add-on
   this.feature = new Feature(variation, studyUtils, REASONS[reason], log);
 
+  // if you have code to handle expiration / long-timers, it could go here
+  /*
+  if (this.feature.hasExpired()) {
+    // Please note that the general study expiration should probably be taken care of by Normandy.
```

Comment:
This is news to me! I have been handling general study expiration in my addon code -- @mythmon , does Normandy handle this? What do you recommend study authors do re: general study expiration?

Reply:
Normandy does not handle this, and I do not agree with this comment.

Normandy will handle removal of the add-on when the study is ended on the server, but that is not what this feature is generally used for. The first expiration timer that I'm aware of was designed to expire an experiment as a fail safe, in case Normandy could not be contacted, or someone forgot to disable the recipe on the server.

In that original case, the expiry timer was not a normal end to the study, but a failsafe in case something went wrong. As such, it should be implemented in a very reliable way by the add-on.

Reply:
Thanks for clarifying @mythmon, adding this information as a comment in the template